### DISCUSSION TITLE
Is it possible to copy all XMP  tags and make it IPTC tags or exif Tags?

### DISCUSSION THREAD

**Discussion by faridelya** [sylikc/pyexiftool#71] [Q&A] [answered]
Hi everyone! @WildEyeConservation @Untouchable17 @dnsamw  @sylikc 
Hope you all doing well. I am stuck when adding a new tag that is not predefined for any metadata standard like
1.  if I copy XMP tags and make them IPTC tags or exif tags?
2. or if I copy   IPTC tags and make them XMP tags or EXIF tags?
4. or If I copy Exif tags and make them XMP tags to IPTC Tags?

currently, I can only add predefine tags for each standard by using this code

This first code working
```
from exiftool import ExifTool
with ExifTool() as et:
    print(et.execute(*["-XMP:Subject=hi"] + ["skyblue.png"]))
```

but the below code not working.
```
from exiftool import ExifTool
with ExifTool() as et:
    print(et.execute(*["-IPTC:Subject=hi"] + ["skyblue.png"]))

```

is there anything wrong with my code or what?, please write your precious answer
Thanks

**Comment by Untouchable17** [comment]
The reason your second code snippet is not working is likely due to the
absence of a corresponding IPTC tag named "Subject" in the image metadata.

In the first code snippet, you are using the -XMP:Subject=hi argument,
which sets the value of the "Subject" tag in the XMP metadata to "hi". This
works because the XMP metadata standard includes a "Subject" tag.

However, in the second code snippet, you are using the -IPTC:Subject=hi
argument, assuming that there is an IPTC tag called "Subject." If this tag
doesn't exist in the image's IPTC metadata, the command will fail to set
the value.

To add IPTC tags, you need to use the correct tag names recognized by the
IPTC standard. The tag names for IPTC metadata may differ from XMP or EXIF.
You should consult the IPTC standard documentation or a reference guide to
find the appropriate IPTC tag names for the desired information you want to
add.

Once you identify the correct IPTC tag name, you can modify your code
accordingly. Here's an example using a hypothetical IPTC tag "Caption":

from exiftool import ExifTool

with ExifTool() as et:
    print(et.execute(*["-IPTC:Caption=SecDet Samurai"] + ["skyblue.png"]))

Make sure to replace "Caption" with the actual IPTC tag name you wish to
set, according to the IPTC standard.

Remember, it's important to use the appropriate tag names for each metadata
standard to ensure compatibility and adherence to the specifications of
each standard.

I hope this clarifies the issue.

вт, 20 июн. 2023 г. в 15:21, Farid Ullah ***@***.***>:

**Reply by sylikc** [reply]
thanks for giving a detailed reply 😍

**Comment by faridelya** [comment]
Hi, @Untouchable17  Thanks for your answer.
so we can say that there is no way to insert tags that are not part of XMP or IPTC or EXIF  standard of metadata.
Only the predefine tags are inserted.

correct me if i am wrong 
thanks

**Reply by sylikc** [reply, marked as answer]
there's no "easy" way... see my reply below.  And for XMP, IPTC, and EXIF, you really shouldn't insert tags which are not part of the specifications... because no other application would read these tags anyways

**Reply by sylikc** [reply]
Also, in terms of how to use exiftool to get custom tags and insert as not part of the standards, it's best to ask this on the exiftool forums, as it's more of an upstream question

**Comment by faridelya** [comment]
hello everyone @WildEyeConservation @Untouchable17 @dnsamw @sylikc 
please answer this very short question that

Can we add tags that are not predefine in each metadata standard by using ExifTool???
if yes please show us the code.

**Reply by sylikc** [reply]
@faridelya There are ways to add custom tags which are not defined by default.  ExifTool has documentation on this ... try googling for many examples of custom tags people have asked https://exiftool.org/forum/index.php?topic=7377.0 and such

If you are set on making custom tags which no other applications read, I have a test which does exactly this, and read/writes binary from an arbitrary tag that I configured in the test.  See https://github.com/sylikc/pyexiftool/commit/60d793f78277816fd73051e388dc9f456ca5ad45

**Reply by faridelya** [reply]
@sylikc Thanks you very much for you kind response
